File: resources/app-mern/src/rest-server/src/interprocess/pythonRedis/redisUtil.py
import redis
import json

from redisKeys import RedisChannel

class RedisUtil:
    def __init__(self):
        self.redis = redis.Redis(host='localhost', port=6379, db=0)
        self.sub = self.redis.pubsub()
    
    def pub(self, channelName, jsonData):
        channel = channelName.value
        self.redis.publish(channel, json.dumps(jsonData))

    def subscribe(self, channelName):
        # get enum value - RoomBPToRoomHumid
        # print(channelName.value)
        channel = channelName.value
        self.sub.subscribe(channel)

File: resources/app-mern/src/rest-server/src/interprocess/nodejsRedis/redisUtil.js
const Redis = require("ioredis");

let RedisUtil = class redisUtil {
    constructor() {
        this.redis = new Redis();
    }

    pub(channelName, jsonData) {
        this.redis.publish(channelName, JSON.stringify(jsonData));
    }
    sub(channelName, callback) {
        this.redis.subscribe(channelName, (err, data) => {
            callback(err, data);
        });
    }
}


module.exports = RedisUtil;